rust/lance/src/io/exec/planner.rs

```diff
@@ -183,25 +203,23 @@ impl ContextProvider for LanceContextProvider {
         )))
     }
 
+    fn get_aggregate_meta(&self, name: &str) -> Option<Arc<AggregateUDF>> {
```

Comment:
`SqlToRel` now relies on the `ContextProvider` for functions (e.g. regexp_match).  These come from a `SessionState` instance and so we needed to add one of those to the `LanceContextProvider` (the default is fine)